src/components/NavBar/Search/Search.component.jsx

```diff
@@ -1,18 +1,55 @@
 import SearchIcon from '@material-ui/icons/Search';
-import React from 'react';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import CancelIcon from '@material-ui/icons/Cancel';
+import { IconButton } from '@material-ui/core';
 import { StyledInput, IconContainer, SearchContainer } from './Search.styled';
 
-const Search = ({ value }) => {
+const Search = ({ handleSearchChange }) => {
+  const history = useHistory();
+  const [keyWord, setKeyWord] = useState('');
+
+  const handleChange = (event) => {
+    setKeyWord(event.target.value);
+  };
+
+  const triggerChange = () => {
+    if (!keyWord) history.push('/');
+    else handleSearchChange(keyWord);
+  };
+
+  const handleKeyDown = (event) => {
+    if (event.key === 'Enter') {
+      triggerChange();
+    }
+  };
+
   return (
     <SearchContainer>
       <IconContainer>
         <SearchIcon />
       </IconContainer>
       <StyledInput
+        id="search-cp"
         placeholder="Search…"
         inputProps={{ 'aria-label': 'search' }}
-        value={value}
+        value={keyWord}
+        onChange={handleChange}
+        onKeyDown={handleKeyDown}
       />
+      {keyWord && (
+        <IconButton size="small">
+          <CancelIcon
+            fontSize="small"
+            color="error"
+            onClick={() => {
+              setKeyWord('');
```

Comment:
this is a pretty nice function! But I think when user click X icon, it should clear the data only. User will type something and enter to search again

Reply:
thats right! I will change the logic a little bit